pkg/daemons/control/server.go

```diff
@@ -347,89 +345,79 @@ func setupStorageBackend(argsMap map[string]string, cfg *config.Control) {
 	}
 }
 
-func cloudControllerManager(ctx context.Context, cfg *config.Control, runtime *config.ControlRuntime) {
-	// Reference: https://github.com/kubernetes/kubernetes/pull/93764
-	// The above-linked change made the in-tree cloud-controller-manager command much more flexible
-	// but much more complicated to wrap. It now validates some of the configuration very early on, before
-	// the CLI args are parsed, so some of the configuration needs to be hardcoded instead of set via flags.
-	// Reference: https://github.com/kubernetes/kubernetes/pull/98210
-	// The above-linked change further clarifies the intent of the example cloud-controller-manager.
-
+func cloudControllerManager(ctx context.Context, cfg *config.Control, runtime *config.ControlRuntime) error {
 	argsMap := map[string]string{
-		"profiling": "false",
-	}
-	args := config.GetArgsList(argsMap, cfg.ExtraCloudControllerArgs)
-
-	ccmOptions, err := ccmopt.NewCloudControllerManagerOptions()
-	if err != nil {
-		logrus.Fatalf("Unable to initialize cloudcontroller options: %v", err)
+		"profiling":                    "false",
```

Comment:
There was some reason I had broken these out and set them directly on the ccmOptions struct; something to do with when upstream was validating the options vs when it parses the flags. If what you're doing works on 1.21 then I'm glad upstream has fixed it (I think that's why they moved to using the `cloudInitializer` func option to `NewCloudControllerManagerCommand` , but you might run into it again when backporting to 1.20.